sdk/src/main/java/io/opentelemetry/sdk/trace/RandomIdsGenerator.java

```diff
@@ -18,23 +18,25 @@
 
 import io.opentelemetry.trace.SpanId;
 import io.opentelemetry.trace.TraceId;
+import java.security.SecureRandom;
 import java.util.Random;
 
 final class RandomIdsGenerator implements IdsGenerator {
-  private static final long INVALID_ID = 0;
-  // TODO change with ThreadLocal version
-  // https://github.com/open-telemetry/opentelemetry-java/issues/406
-  private final Random random;
 
-  RandomIdsGenerator(Random random) {
-    this.random = random;
-  }
+  private static final ThreadLocal<Random> threadLocalRandom =
+      new ThreadLocal<Random>() {
```

Comment:
This may be a bit too much memory, do you know how much memory is used by an instance of the SecureRandom? Also what is the reason to use SecureRandom vs just Random?

Reply:
I had only used it because that's what was being passed in by default in the TracerSdkFactory. In practice, I honestly don't think that we need the "true" randomness that SecureRandom will provide, and Random should work totally fine for generating trace/span ids.  And, since it doesn't depend on a source of entropy, it will be much less likely to block under high load. I'm very happy to switch to bog-standard Random for this.

Reply:
Also, I don't think that a Random instance per thread is going to make much of a memory impact. Random itself only holds onto the seed as an AtomicLong, so very little memory overhead.

Reply:
I checked, and it is not making a big difference for memory.